src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +610,447 @@ def paramsToList(params, config) {
+  paramSets = paramSets.collect({ paramValues ->
+    [paramValues.get("id", null), paramValues.findAll{it.key != 'id'}]
+  })
+  // Split parameters with 'multiple: true'
+  paramSets = paramSets.collect({ id, paramValues ->
+    def splitParamValues = _splitParams(paramValues, config)
+    [id, splitParamValues]
+  })
+  
+  // The paths of input files inside a param_list file may have been specified relatively to the
+  // location of the param_list file. These paths must be made absolute.
+  if (paramListFile){
+    paramSets = paramSets.collect({ id, paramValues ->
+      def relativeParamValues = _resolvePathsRelativeTo(paramValues, config, paramListFile)
+      [id, relativeParamValues]
+    })
+  }
+
+  return paramSets
+}
+
+/**
+ * Cast parameters to the correct type as defined in the Viash config
+ *
+ * @param parValues: a Map of input arguments.
+ *
+ * @return the input arguments that have been cast to the type from the viash config.
+ */
+
+private Map<String, Object> _castParamTypes(Map<String, Object> parValues, Map config) {
+  // Cast the input to the correct type according to viash config
+  def castParValues = parValues.collectEntries({ parName, parValue ->
+    paramSettings = config.functionality.allArguments.find({it.plainName == parName})
+    // dont parse parameters like publish_dir ( in which case paramSettings = null)
+    parType = paramSettings ? paramSettings.get("type", null) : null
+    if (parValue !instanceof Collection) {
+      parValue = [parValue]
+    }
+    if (parType == "file" && ((paramSettings.direction ?: "input") == "input")) {
+      parValue = parValue.collect{ path ->
+        if (path !instanceof String) {
+          path
+        } else {
+          file(path)
+        }
+      }
+    } else if (parType == "integer") {
+      parValue = parValue.collect{it as Integer}
+    } else if (parType == "double") {
+      parValue = parValue.collect{it as Double}
+    } else if (parType == "boolean" || 
+                parType == "boolean_true" || 
+                parType == "boolean_false") {
+      parValue = parValue.collect{it as Boolean}
+    }
+
+    // simplify list to value if need be
+    if (paramSettings && !paramSettings.multiple) {
+      assert parValue.size() == 1 : 
+        "Error: argument ${parName} has too many values.\n" +
+        "  Expected amount: 1. Found: ${parValue.size()}"
+      parValue = parValue[0]
+    }
+    [parName, parValue]
+  })
+  return castParValues
+}
+
+/**
+ * Apply the argument settings specified in a Viash config to a single parameter set.
+ *    - Split the parameter values according to their seperator if 
+ *       the parameter accepts multiple values
+ *    - Cast the parameters to their corect types.
+ *    - Assertions:
+ *        ~ Check if any unknown parameters are found
+ * 
+ * @return The input parameters that have been processed.
+ */
+private Map<String, Object> applyConfigToOneParameterSet(Map<String, Object> paramValues, Map config){
```

Comment:
This assumes paramValues has already been processed, right? (because kebab-case arguments will turn into camelCase and vice versa).

Can you add javadocs for the paramValues and config?

Reply:
Yes, it assumes that `paramValues` contains only parameters that are in the viash config.